lib/Partitioner/Partitioner.cpp

```diff
@@ -378,11 +411,37 @@ llvm::Expected<DAGListTy> Partitioner::createDAGWithoutPartition(
   return std::move(partitions);
 }
 
-llvm::Error Partitioner::loadBalancedPartitioning(Function *F,
-                                                  DeviceIDTy numDevices,
-                                                  uint64_t availableMemory,
-                                                  llvm::StringRef backendName,
-                                                  NodeToFunctionMap &mapping) {
+llvm::Expected<DAGListTy>
+Partitioner::loadBalancedPartition(CompilationContext &cctx,
+                                   size_t numDevices) {
+  DCHECK(module_->getFunctions().size() == 1)
```

Comment:
as talked offline, similarly to the rest of the cases when we return error, we should return error here as well instead of debug level checking.